packages/default-theme/components/SwTopNavigation.vue

```diff
@@ -89,46 +85,60 @@ import {
   useProductSearch
 } from '@shopware-pwa/composables'
 import SwLoginModal from '@shopware-pwa/default-theme/components/modals/SwLoginModal'
+import SwMegaMenu from '@shopware-pwa/default-theme/components/SwMegaMenu'
+import { ref, reactive, onMounted } from "@vue/composition-api";
+import { getNavigation } from '@shopware-pwa/shopware-6-client'
 import { PAGE_ACCOUNT } from '@shopware-pwa/default-theme/helpers/pages'
 
 export default {
-  components: { SfHeader, SfCircleIcon, SwLoginModal, SfImage, SfSearchBar },
+  components: { SfHeader, SfCircleIcon, SwLoginModal, SfImage, SfSearchBar, SwMegaMenu },
   setup() {
-    const { routes, fetchRoutes } = useNavigation()
     const { isLoggedIn, logout } = useUser()
     const { count } = useCart()
     const { toggleSidebar } = useCartSidebar()
     const { toggleModal } = useUserLoginModal()
     const { search: fulltextSearch } = useProductSearch()
 
+    const navigationElements = reactive([])
+    const hoveredNavigationItem = ref('')
+
+    onMounted(async () => {
+      const { children } = await getNavigation({
```

Comment:
Is it possible to move the logic into `useNavigation` composable?

Reply:
sure, good idea!